Allow customization of the controller used in component tests (#465)
github/view_component: docs/index.md

config.view_component.preview_controller = "MyPreviewController"
```

### Configuring the controller used in tests

Component tests assume the existence of an `ApplicationController` class, which can be configured globally using the `test_controller` option:

```ruby
config.view_component.test_controller = "BaseController"
```

> You can configure the controller used for a test case using the `with_controller_class` helper from `ViewComponent::TestHelpers`.

Reviewer: ```suggestion
To configure the controller used for a specific test case, use `with_controller_class` from `ViewComponent::TestHelpers`:
```

```ruby
# Minitest
class ExampleComponentTest < ViewComponent::TestCase
  def test_component_in_public_controller
    with_controller_class PublicController do
      render_inline ExampleComponent.new

      assert_text "foo"
    end
  end

  def test_component_in_authenticated_controller
    with_controller_class AuthenticatedController do
      render_inline ExampleComponent.new

      assert_text "bar"
    end
  end
end

> 
> # RSpec
> RSpec.describe ExampleComponent, type: :component do
>   context "with public controller" do
>     around { |example| with_controller_class(PublicController, &example) }
> 
>     it "shows foo" do
>       render_inline ExampleComponent.new
> 
>       expect(page).to have_text("foo")
>     end
>   end
> 
>   context "with authenticated controller" do
>     around { |example| with_controller_class(AuthenticatedController, &example) }
> 
>     it "shows bar" do
>       render_inline ExampleComponent.new
> 
>       expect(page).to have_text("bar")
>     end
>   end
> end

Reviewer: While we happily support RSpec, I'm hesitant to have examples from multiple test frameworks in the documentation.

```suggestion
```
